src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -204,6 +204,10 @@ internal void DispatchEventRecord(TraceEventNativeMethods.EVENT_RECORD* eventRec
                 Debug.Assert(sessionEndTimeQPC == 0 || eventRecord->EventHeader.TimeStamp - sessionEndTimeQPC < _QPCFreq * 24 * 3600);
 
                 var traceEvent = Lookup(eventRecord);
+                if (traceEvent.NeedsFixup)
```

Comment:
>NeedsFixup [](http://example.com/codeflow?start=31&length=10)

perhaps this should be called IsDyanmicEvent instead of NeedsFixup

Reply:
`NeedsFixup` is a TraceEvent-level construct for events that require field updates/modifications prior to dispatch.  It's not something specific to this PR.